### DISCUSSION TITLE
Compile with NDK in Android

### DISCUSSION THREAD

**Discussion by Sakurawy** [nlohmann/json#2901] [Q&A] [answered]
Hi, everyone!
I've got problems when compiling with ndk in the Andriod project. It occurs as soon as  I include <json.hpp>. The error message is as follows.

error: static_assert failed due to requirement 'sizeof(nlohmann::basic_json<std::map, std::vector, std::__ndk1::basic_string<char, std::__ndk1::char_traits<char>, std::__ndk1::allocator<char> >, bool, long, unsigned long, double, std::allocator, adl_serializer>) == 0' "could not find from_json() method in T's namespace"

Any help? 
Thanks

**Comment by nlohmann** [comment, marked as answer]
That error should be unrelated to Android, but is rather a problem in a missing conversion function. Can you share your code please?

**Reply by Sakurawy** [reply]
Thanks for your reply. The code snippet with json is here:
```c++
using json = nlohmann::json; 
json data = {                                                                                                    
                     {"data", {                                                                                           
                                  {"status", 2},                                                                          
                                  {"format", this->DATA.format},                                                          
                                  {"encoding", this->DATA.encoding},                                                      
                                  {"audio", get_base64_encode(std::string((char *)buffer, len))},                                                       
                              }}};                                                                                        
                                                                                                                          
m_endpoint->send(m_hdl, data.dump(), websocketpp::frame::opcode::text);    
```

**Reply by nlohmann** [reply]
What are the types of this->DATA.format and this->DATA.encoding?
